sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -1,20 +1,742 @@
+    try {
+      long blockSize = getBlockSizeLong(stat);
+      long totalBlocks = getBlockCountLong(stat);
+      return totalBlocks * blockSize;
+    } catch (Exception e) {
+      log(SentryLevel.ERROR, "Error getting total external storage amount.", e);
+      return null;
+    }
+  }
+
+  private boolean isExternalStorageMounted() {
+    return Environment.getExternalStorageState().equals(Environment.MEDIA_MOUNTED)
+        && !Environment.isExternalStorageEmulated();
+  }
+
+  /**
+   * Get the unused amount of external storage, in bytes, or null if no external storage is mounted.
+   *
+   * @return the unused amount of external storage, in bytes, or null if no external storage is
+   *     mounted
+   */
+  private Long getUnusedExternalStorage(StatFs stat) {
+    try {
+      long blockSize = getBlockSizeLong(stat);
+      long availableBlocks = getAvailableBlocksLong(stat);
+      return availableBlocks * blockSize;
+    } catch (Exception e) {
+      log(SentryLevel.ERROR, "Error getting unused external storage amount.", e);
+      return null;
+    }
+  }
+
+  /**
+   * Get the DisplayMetrics object for the current application.
+   *
+   * @return the DisplayMetrics object for the current application
+   */
+  private DisplayMetrics getDisplayMetrics() {
+    try {
+      return context.getResources().getDisplayMetrics();
+    } catch (Exception e) {
+      log(SentryLevel.ERROR, "Error getting DisplayMetrics.", e);
+      return null;
+    }
+  }
+
+  private OperatingSystem getOperatingSystem() {
+    OperatingSystem os = new OperatingSystem();
+    os.setName("Android");
+    os.setVersion(Build.VERSION.RELEASE);
+    os.setBuild(Build.DISPLAY);
+    os.setKernelVersion(getKernelVersion());
+    os.setRooted(isRooted());
+    //    os.setRawDescription("");  whats that? maybe the const of VERSION_CODES?
+
+    return os;
+  }
+
+  private App getApp(PackageInfo packageInfo) {
+    App app = new App();
+    app.setIdentifier(packageInfo.packageName);
+    app.setStartTime(
+        DateUtils
+            .getCurrentDateTime()); // is it App. start? this was not supposed to be a lazy class?
+    //    app.setHash(); whats that?
+    //    app.setBuildType(); whats that? might be possible with BuildConfig.BUILD_VARIANT but Apps
+    // side, also for flavor
+    app.setName(getApplicationName());
+    app.setVersion(packageInfo.versionName);
+    app.setBuild(getVersionCode(packageInfo));
+
+    // App name and starttime don't depend on packageInfo, should we move outside of the packageInfo
+    // NPE check like the old the SDK?
+
+    return app;
+  }
+
+  /**
+   * Get the device's current kernel version, as a string. Attempts to read /proc/version, and falls
+   * back to the 'os.version' System Property.
+   *
+   * @return the device's current kernel version, as a string
+   */
+  private String getKernelVersion() {
+    // its possible to try to execute 'uname' and parse it or also another unix commands or even
```

Comment:
expensive stuff like this should be cached on first use. Since we have multiple hubs. Maybe this could be done in a class with a static field that holds the value and a double check lock for reading/initializing?

Reply:
I think not only what is heavy, but also everything that would not change, let's do it in a different PR